package_test.go

```diff
@@ -581,6 +581,80 @@ func (s *PackageSuite) TestAllErrors(c *C) {
 	}
 }
 
+func (s *PackageSuite) TestTransactions(c *C) {
+	dropTables, sqldb, err := personAndAddressDB()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	selectStmt := sqlair.MustPrepare("SELECT &Person.* FROM person WHERE address_id = $Person.address_id", Person{})
+	insertStmt := sqlair.MustPrepare("INSERT INTO person VALUES ( $Person.name, $Person.id, $Person.address_id, 'fred@email.com');", Person{})
+	var derek = Person{ID: 85, Fullname: "Derek", PostalCode: 8000}
+
+	db := sqlair.NewDB(sqldb)
+	tx, err := db.Begin(nil, nil)
+	if err != nil {
+		c.Fatal(err)
+	}
+	// Insert derek then rollback.
+	q := tx.Query(nil, insertStmt, &derek)
+	// TODO: replace this with q.Run() or equivlent.
+	iter := q.Iter()
+	iter.Next()
+	err = iter.Close()
+	if err != nil {
+		c.Fatal(err)
+	}
+	err = tx.Rollback()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Check derek isnt in db, insert derek; commit.
+	tx, err = db.Begin(nil, nil)
+	if err != nil {
+		c.Fatal(err)
+	}
+	var derekCheck = Person{}
+	err = tx.Query(nil, selectStmt, &derek).One(&derekCheck)
+	if !errors.Is(err, sqlair.ErrNoRows) {
+		c.Fatalf("test failed, got err %s, expected %s", err, sqlair.ErrNoRows)
+	}
+	q = tx.Query(nil, insertStmt, &derek)
+	// TODO: replace this with q.Run() or equivlent.
+	iter = q.Iter()
+	iter.Next()
+	err = iter.Close()
+	if err != nil {
+		c.Fatal(err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Check derek is now in the db.
+	tx, err = db.Begin(nil, nil)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	err = tx.Query(nil, selectStmt, &derek).One(&derekCheck)
+	if err != nil {
+		c.Fatal(err)
+	}
+	c.Assert(derek, Equals, derekCheck)
+	err = tx.Commit()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	_, err = db.Unwrap().Exec(dropTables)
```

Comment:
We should now be using our own implementation instead of unwrapping it. Let's get used to our own API. :)

You'll probably start wishing for something like Query.Run(), which again is the sort of reason why we want to dogfood it.

Reply:
Ah yea, I had already replaced these with Query.Run in the other one, I was waiting until that was available first from the other PR.

Reply:
Changed.